File: back/src/main/java/com/tdevilleduc/urthehero/back/service/IPageService.java
package com.tdevilleduc.urthehero.back.service;

import com.tdevilleduc.urthehero.back.model.Page;
import com.tdevilleduc.urthehero.back.model.PageDTO;

import java.util.Optional;

public interface IPageService {

    boolean exists(Integer pageId);
    boolean notExists(Integer pageId);
    Optional<Page> findById(Integer pageId);
    PageDTO createOrUpdate(PageDTO page);
    void delete(Integer pageId);
}

File: back/src/main/java/com/tdevilleduc/urthehero/back/service/IPageService.kt
package com.tdevilleduc.urthehero.back.service

import com.tdevilleduc.urthehero.back.model.Page
import com.tdevilleduc.urthehero.back.model.PageDTO
import java.util.*

interface IPageService {
    fun exists(pageId: Int): Boolean
    fun notExists(pageId: Int): Boolean
    fun findById(pageId: Int): Page
    fun createOrUpdate(pageDto: PageDTO): PageDTO
    fun delete(pageId: Int)
    fun countByStoryId(storyId: Int): Long
}